internal/run/convert.go

```diff
@@ -0,0 +1,104 @@
+package run
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	convertcmd "github.com/helm/helm-2to3/cmd"
+	"github.com/helm/helm-2to3/pkg/common"
+	"github.com/pelotech/drone-helm3/internal/env"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/storage/driver"
+)
+
+func v3ReleaseFound(release string, cfg *action.Configuration) bool {
+
+	// check if release exists
+	_, err := cfg.Releases.Deployed(release)
+	switch {
+	// case errors.Is(err, driver.ErrReleaseNotFound), errors.Is(err, driver.ErrNoDeployedReleases):
+	case errors.Is(err, driver.ErrReleaseNotFound):
+		log.Printf("No v3 Release of %s found", release)
+	case err == nil:
+		log.Printf("A v3 Release of %s was found", release)
+		return true
+	}
+
+	return false
+}
+
+// Convert holds the parameters to run the Convert action
+type Convert struct {
+	namespace      string
+	debug          action.DebugLog
+	kubeConfig     string
+	kubeContext    string
+	convertOptions convertcmd.ConvertOptions
+}
+
+// NewConvert initialize Convert by using values from env.Config
+func NewConvert(cfg env.Config, kubeConfig string, kubeContext string) *Convert {
+
+	convert := &Convert{
+		namespace:   cfg.Namespace,
+		kubeConfig:  kubeConfig,
+		kubeContext: kubeContext,
+	}
+
+	if cfg.ReleaseVersionsMax == 0 {
+		cfg.ReleaseVersionsMax = 10
+	}
+
+	convert.convertOptions = convertcmd.ConvertOptions{
+		DeleteRelease:      cfg.DeleteV2Releases,
+		DryRun:             cfg.DryRun,
+		MaxReleaseVersions: cfg.ReleaseVersionsMax,
```

Comment:
**nit:**

```suggestion
		MaxReleaseVersions: cfg.MaxReleaseVersions,
```

Reply:
makes sense. The interesting part if that the cli option is called `--release-versions-max`, so I was matching that before.